stdlib/xml/dom/expatbuilder.pyi

```diff
@@ -1,3 +1,103 @@
+# NOTE: Incomplete, lots of unannotated parameters currently
 
-def __getattr__(name: str) -> Any: ...  # incomplete
+from typing import Any, NoReturn
+from typing_extensions import Literal
+from xml.dom.minidom import Document, DOMImplementation, Node, TypeInfo
+from xml.dom.xmlbuilder import DOMBuilderFilter, Options
+
+TEXT_NODE = Node.TEXT_NODE
+CDATA_SECTION_NODE = Node.CDATA_SECTION_NODE
+DOCUMENT_NODE = Node.DOCUMENT_NODE
+FILTER_ACCEPT = DOMBuilderFilter.FILTER_ACCEPT
+FILTER_REJECT = DOMBuilderFilter.FILTER_REJECT
+FILTER_SKIP = DOMBuilderFilter.FILTER_SKIP
+FILTER_INTERRUPT = DOMBuilderFilter.FILTER_INTERRUPT
+theDOMImplementation: DOMImplementation | None
+
+class ElementInfo:
+    tagName: Any
+    def __init__(self, tagName, model: Any | None = ...) -> None: ...
+    def getAttributeType(self, aname) -> TypeInfo: ...
+    def getAttributeTypeNS(self, namespaceURI, localName) -> TypeInfo: ...
+    def isElementContent(self) -> bool: ...
+    def isEmpty(self) -> bool: ...
+    def isId(self, aname) -> bool: ...
+    def isIdNS(self, euri, ename, auri, aname) -> bool: ...
+
+class ExpatBuilder:
+    doc: Document
+    curnode: Document
+    def __init__(self, options: Options | None = ...) -> None: ...
+    def createParser(self): ...
+    def getParser(self) -> Any | None: ...
+    def reset(self) -> None: ...
+    def install(self, parser) -> None: ...
+    def parseFile(self, file) -> Document: ...
+    def parseString(self, string: str) -> Document: ...
+    def start_doctype_decl_handler(self, doctypeName, systemId, publicId, has_internal_subset) -> None: ...
+    def end_doctype_decl_handler(self) -> None: ...
+    def pi_handler(self, target, data) -> None: ...
+    def character_data_handler_cdata(self, data) -> None: ...
+    def character_data_handler(self, data) -> None: ...
+    def start_cdata_section_handler(self) -> None: ...
+    def end_cdata_section_handler(self) -> None: ...
+    def entity_decl_handler(self, entityName, is_parameter_entity, value, base, systemId, publicId, notationName) -> None: ...
+    def notation_decl_handler(self, notationName, base, systemId, publicId) -> None: ...
+    def comment_handler(self, data) -> None: ...
+    def external_entity_ref_handler(self, context, base, systemId, publicId) -> Literal[1]: ...
+    def first_element_handler(self, name, attributes) -> None: ...
+    def start_element_handler(self, name, attributes) -> None: ...
+    def end_element_handler(self, name) -> None: ...
+    def element_decl_handler(self, name, model) -> None: ...
+    def attlist_decl_handler(self, elem, name, type, default, required) -> None: ...
+    def xml_decl_handler(self, version, encoding, standalone) -> None: ...
+
+class FilterVisibilityController:
+    filter: DOMBuilderFilter
+    def __init__(self, filter: DOMBuilderFilter) -> None: ...
+    def startContainer(self, node: Node) -> Literal[1, 2, 3]: ...
+    def acceptNode(self, node: Node) -> Literal[1, 2, 3]: ...
```

Comment:
Maybe these should be just `int`. It isn't easy to tell what `1`, `2` and `3` represent.

Reply:
Yeah, fair.